libcloud/compute/base.py

```diff
@@ -1316,7 +1320,7 @@ def filter_addresses(addresses):
         uuids = set([node.uuid for node in nodes])
 
         while time.time() < end:
-            all_nodes = self.list_nodes()
+            all_nodes = self.list_nodes(ex_cloud_opts)
```

Comment:
It would be better to make this argument a dict - this way it's more generic and user can pass multiple arguments to the list method. And we should also call it `ex_list_nodes_kwargs` to make it more explicit.

Then we would do something like that:

```python
...
all_nodes = self.list_nodes(**ex_list_nodes_kwargs)
```